File: frontend/src/domains/Usuario.js
import { observable } from 'mobx';
import DomainBase from './DomainBase';

class Usuario extends DomainBase {
  @observable id;
  @observable nome;
  @observable email;
  @observable nomeCanal;
  @observable dataNascimento;

  static getDomainAttributes() {
    return ['id', 'nome', 'email', 'nomeCanal', 'dataNascimento'];
  }
}

export default Usuario;

File: backend/src/main/java/br/gov/ac/tce/licon/entities/Usuario.java
package br.gov.ac.tce.licon.entities;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;


@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Usuario extends AbstractAuditableEntity {


	@NotNull
	@Column(name = "NOME")
	private String nome;

	@NotNull
	@Column(name = "EMAIL")
	private String email;

	@NotNull
	@Column(name = "NOME_CANAL")
	private String nomeCanal;

	@NotNull
	@Column(name = "DATA_NASCIMENTO")
	private LocalDate dataNascimento;

}
